calico_containers/tests/unit/checksystem_test.py

```diff
@@ -69,3 +71,99 @@ def m_module_loaded(module):
+        :param ip_forwarding_status: return_value for _check_ip_forwarding
+        :param docker_version_status: return_value for _check_docker_version
+        """
+        with \
+                patch('calico_ctl.checksystem.enforce_root', autospec=True) \
+                        as m_enforce_root, \
+                patch('calico_ctl.checksystem._check_kernel_modules', autospec=True) \
+                        as m_check_kernel_modules, \
+                patch('calico_ctl.checksystem._check_ip_forwarding', autospec=True) \
+                        as m_check_ip_forwarding, \
+                patch('calico_ctl.checksystem._check_docker_version', autospec=True) \
+                        as m_check_docker_version:
+            # Set up mock objects
+            m_check_kernel_modules.return_value = kernel_status
+            m_check_ip_forwarding.return_value = ip_forwarding_status
+            m_check_docker_version.return_value = docker_version_status
+
+            # Call method under test without exiting if error detected
+            test_return = check_system(fix=False, quit_if_error=False)
+
+            # Assert
+            m_enforce_root.assert_called_once_with()
+            self.assertFalse(test_return)
+
+    @parameterized.expand([
+        (True, True, False),
+        (True, False, True),
+        (False, True, True),
+    ])
+    def test_check_system_bad_state_quit(
+            self, kernel_status, ip_forwarding_status, docker_version_status):
+        """
+        Test for check_system when one of the system checks fails
+
+        This test exits if there is a detected error -
+        Assert that the system exits
+
+        :param kernel_status: return_value for _check_kernel_modules patch
+        :param ip_forwarding_status: return_value for _check_ip_forwarding patch
+        :param docker_version_status: return_value for _check_docker_version patch
+        """
+        with \
+                patch('calico_ctl.checksystem.enforce_root', autospec=True) \
+                        as m_enforce_root, \
+                patch('calico_ctl.checksystem._check_kernel_modules', autospec=True) \
+                        as m_check_kernel_modules, \
+                patch('calico_ctl.checksystem._check_ip_forwarding', autospec=True) \
+                        as m_check_ip_forwarding, \
+                patch('calico_ctl.checksystem._check_docker_version', autospec=True) \
+                        as m_check_docker_version:
+            # Set up mock objects
+            m_check_kernel_modules.return_value = kernel_status
+            m_check_ip_forwarding.return_value = ip_forwarding_status
+            m_check_docker_version.return_value = docker_version_status
+
+            # Call method under test expecting a SystemExit when fail detected
+            self.assertRaises(SystemExit, check_system, fix=False, quit_if_error=True)
+
+            # Assert
+            m_enforce_root.assert_called_once_with()
```

Comment:
When I first suggested editing this file, all of the `check_kernel_modules`, `check_ip_forwarding`, and `check_docker_version` function code were within the `check_system`function.  These functions are the real meat of the check_system file that should be tested.  We should add tests for these or put it on the backlog.